Lib/random.py

```diff
@@ -94,6 +94,18 @@ def __init__(self, x=None):
         self.seed(x)
         self.gauss_next = None
 
+    def __init_subclass__(cls, **kwargs):
+        # Only call self.getrandbits if the original random() builtin method
+        # has not been overridden or if a new getrandbits() was supplied.
+        if type(cls.__dict__.get('getrandbits')) is _FunctionType:
```

Comment:
Why not just `'getrandbits' in cls.__dict__`?

Reply:
to preserve the existing behavior as much as possible; if getrandbits gets overridden with an attribute instead of a method, the fallback mechanism gets activated

Reply:
We already change the behavior by allowing getrandbits() to win even if random() is overridden in a parent. I think it would be better to allow overriding getrandbits() not only with Python functions.

Reply:
meaning that when you override it with something that isn't callable _randbelow will fail with an exception; it doesn't do that now, but I agree it may actually be the right thing to do because such a situation would probably only indicate a mistake.